app/models/ability.rb

```diff
@@ -10,11 +10,17 @@ def initialize(user)
     #   can :read, :all
     #   return unless user.admin?
 
-    can(%i[enroll read read mark_as], :all)
+    can(%i[enroll read mark_as], :all) if user.member?
 
     can(:manage, Course, user_id: user.id) if user.admin?
 
-    can(%i[update], Course, superuser_id: user.id)
+    can(:update, Course, superuser_id: user.id)
+
+    can(%i[read mark_as enroll], Course)
```

Comment:
@amnindersingh12 While defining abilities:
the ones with less restrictions should come at top and the one more with more restrictions should go at end

For example:
```
    can :read, Post, public: true

    return unless user.present?  # additional permissions for logged in users (they can read their own posts)
    can :read, Post, user: user

    return unless user.admin?  # additional permissions for administrators
    can :read, Post # admin can read all posts by all users
```